crossdc-producer/src/main/java/org/apache/solr/update/processor/MirroringUpdateProcessor.java

```diff
@@ -127,58 +125,82 @@ private UpdateRequest createAndOrGetMirrorRequest() {
+    if (doMirroring && !cmd.isDeleteById() && !"*:*".equals(cmd.query)) {
+      CloudDescriptor cloudDesc = cmd.getReq().getCore().getCoreDescriptor().getCloudDescriptor();
       String collection = cloudDesc.getCollectionName();
-
       HttpClient httpClient = cmd.getReq().getCore().getCoreContainer().getUpdateShardHandler().getDefaultHttpClient();
 
-      try (HttpSolrClient client =
-          new HttpSolrClient.Builder(cmd.getReq().getCore().getCoreContainer().getZkController().getBaseUrl()).withHttpClient(httpClient).build()) {
-
+      try (HttpSolrClient client = new HttpSolrClient.Builder(cmd.getReq().getCore().getCoreContainer().getZkController().getBaseUrl()).withHttpClient(httpClient).build()) {
         String uniqueField = cmd.getReq().getSchema().getUniqueKeyField().getName();
 
-        int rows = Integer.getInteger("solr.crossdc.dbq_rows", 1000);
-        SolrQuery q = new SolrQuery(cmd.query).setRows(rows).setSort(SolrQuery.SortClause.asc(uniqueField)).setFields(uniqueField);
-        String cursorMark = CursorMarkParams.CURSOR_MARK_START;
-
-        int cnt = 1;
-        boolean done = false;
-        while (!done) {
-          q.set(CursorMarkParams.CURSOR_MARK_PARAM, cursorMark);
-          QueryResponse rsp =
-              client.query(collection, q);
-          String nextCursorMark = rsp.getNextCursorMark();
-
-          if (log.isDebugEnabled()) {
-            log.debug("resp: cm={}, ncm={}, cnt={}, results={} ", cursorMark, nextCursorMark, cnt,
-                rsp.getResults());
-            cnt++;
+        if (dbqMethod == null || dbqMethod.equals("default")) {
+
+          int rows = Integer.getInteger("solr.crossdc.dbq_rows", 1000);
+          SolrQuery q = new SolrQuery(cmd.query).setRows(rows).setSort(SolrQuery.SortClause.asc(uniqueField)).setFields(uniqueField);
+          String cursorMark = CursorMarkParams.CURSOR_MARK_START;
+
+          int cnt = 1;
+          boolean done = false;
+          while (!done) {
+            q.set(CursorMarkParams.CURSOR_MARK_PARAM, cursorMark);
+            QueryResponse rsp =
+                    client.query(collection, q);
+            String nextCursorMark = rsp.getNextCursorMark();
+
+            if (log.isDebugEnabled()) {
+              log.debug("resp: cm={}, ncm={}, cnt={}, results={} ", cursorMark, nextCursorMark, cnt,
+                      rsp.getResults());
+              cnt++;
+            }
+
+            processDBQResults(client, collection, uniqueField, rsp);
+            if (cursorMark.equals(nextCursorMark)) {
+              done = true;
+            }
+            cursorMark = nextCursorMark;
           }
-
-          processDBQResults(client, collection, uniqueField, rsp);
-          if (cursorMark.equals(nextCursorMark)) {
-            done = true;
+        } else if (dbqMethod.equals("convert_no_paging")) {
+          int rows = 10000;
+          SolrQuery q = new SolrQuery(cmd.query).setRows(rows).setFields(uniqueField);
```

Comment:
[Q] I wonder if we could use a higher rows value if we fetched the documents using the /export handler, rather than /select?

Or even better: a `delete(..., search(...))` streaming expression would use the /export handler implicitly, and batch and send the delete-by-ID requests for us.  It looks like that `delete` streaming expression was built with this specific use case in mind. (see [here](https://issues.apache.org/jira/browse/SOLR-14241) and [here](https://solr.apache.org/guide/8_7/stream-decorator-reference.html#delete) )

Reply:
I think that could make a good additional option, but given it's an optional endpoint, I don't think it should replace a built in option.